packages/multichain/src/scope/transform.test.ts

```diff
@@ -0,0 +1,352 @@
+        ...validScopeObject,
+        rpcDocuments: ['a', 'b', 'c'],
+      });
+    });
+
+    it('returns an object with the unique set of rpcEndpoints', () => {
+      expect(
+        mergeScopeObject(
+          {
+            ...validScopeObject,
+            rpcEndpoints: ['a', 'b', 'c'],
+          },
+          {
+            ...validScopeObject,
+            rpcEndpoints: ['b', 'c', 'd'],
+          },
+        ),
+      ).toStrictEqual({
+        ...validScopeObject,
+        rpcEndpoints: ['a', 'b', 'c', 'd'],
+      });
+
+      expect(
+        mergeScopeObject(
+          {
+            ...validScopeObject,
+            rpcEndpoints: ['a', 'b', 'c'],
+          },
+          {
+            ...validScopeObject,
+          },
+        ),
+      ).toStrictEqual({
+        ...validScopeObject,
+        rpcEndpoints: ['a', 'b', 'c'],
+      });
+
+      expect(
+        mergeScopeObject(
+          {
+            ...validScopeObject,
+          },
+          {
+            ...validScopeObject,
+            rpcEndpoints: ['a', 'b', 'c'],
+          },
+        ),
+      ).toStrictEqual({
+        ...validScopeObject,
+        rpcEndpoints: ['a', 'b', 'c'],
+      });
+    });
+  });
+
+  describe('mergeScopes', () => {
+    it('merges the scopeObjects with matching scopeString', () => {
+      expect(
+        mergeScopes(
+          {
+            'eip155:1': {
+              methods: ['a', 'b', 'c'],
+              notifications: ['foo'],
+              accounts: [],
+            },
+          },
+          {
+            'eip155:1': {
+              methods: ['c', 'd'],
+              notifications: ['bar'],
+              accounts: [],
+            },
+          },
+        ),
+      ).toStrictEqual({
+        'eip155:1': {
+          methods: ['a', 'b', 'c', 'd'],
+          notifications: ['foo', 'bar'],
+          accounts: [],
+        },
+      });
+    });
+
+    it('preserves the scopeObjects with no matching scopeString', () => {
+      expect(
+        mergeScopes(
+          {
+            'eip155:1': {
+              methods: ['a', 'b', 'c'],
+              notifications: ['foo'],
+              accounts: [],
+            },
+          },
+          {
+            'eip155:2': {
+              methods: ['c', 'd'],
+              notifications: ['bar'],
+              accounts: [],
+            },
+            'eip155:3': {
+              methods: [],
+              notifications: [],
+              accounts: [],
+            },
+          },
+        ),
+      ).toStrictEqual({
+        'eip155:1': {
+          methods: ['a', 'b', 'c'],
+          notifications: ['foo'],
+          accounts: [],
+        },
+        'eip155:2': {
+          methods: ['c', 'd'],
+          notifications: ['bar'],
+          accounts: [],
+        },
+        'eip155:3': {
+          methods: [],
+          notifications: [],
+          accounts: [],
+        },
+      });
+    });
+  });
+
+  describe('normalizeAndMergeScopes', () => {
```

Comment:
Nit: Additional test case ideas:
* Empty object as input
* Object that is already normalized

Reply:
added here: https://github.com/MetaMask/core/pull/4784/commits/7ceab946cc4829dca983d577a00ca37623b55001